crates/sui-types/src/sui_system_state.rs

```diff
@@ -586,9 +702,26 @@ where
         .data
         .try_as_move()
         .ok_or(SuiError::SuiSystemStateNotFound)?;
-    let result = bcs::from_bytes::<Field<u64, SuiSystemState>>(move_object.contents())
-        .expect("Sui System State object deserialization cannot fail");
-    Ok(result.value)
+    match wrapper.version {
+        1 => {
+            let result =
+                bcs::from_bytes::<Field<u64, SuiSystemStateInnerV1>>(move_object.contents())
+                    .expect("Sui System State object deserialization cannot fail");
+            Ok(SuiSystemState::V1(result.value))
+        }
+        // The following case is for sim_test only to support authority_tests::test_sui_system_state_nop_upgrade.
+        #[cfg(msim)]
+        2 => {
```

Comment:
just FYI this will obviously break after we add the "real" version 2. We can punt on this but also feel free to copy the pattern I used here: https://github.com/MystenLabs/sui/blob/main/crates/sui-protocol-config/src/lib.rs#L28-L33 for dealing with this